### ISSUE TITLE
Errors in building pupil_detector module

### ISSUE THREAD

**Issue by injoonhong** [pupil-labs/pupil#822]
Hi all, 

I tried to build pupil_detector module explicitly by running `python setup.py build`  but run into the following error.

fatal error C1047: The object or library file 'build\temp.win-amd64-3.6\Release\detector_3d.obj' was created with an older compiler than other objects; rebuild old objects and libraries
LINK : fatal error LNK1257: code generation failed
error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\BIN\\x86_amd64\\link.exe' failed with exit status 1257

It says I compiled detector_3d.obj using older compiler than other objects, but I do now know how to resolve this problem.

My development environment is Window with Visual Studio 2017. 
FYI,  the other library "detector_2d.cp36-win_amd64.lib" had been successfully built right before the 3d library build error was coming up. 

Does anyone know how to resolve the problem?
Looking forward to your kind advise.

Thank you.

Regards,
Injoon

**Comment by willpatera** [comment]
@injoonhong You should be using VS 2017 preview. Detector 3d relies on the ceres library, so it is possible that you were not able to successfully build the 3d detector due to issues related to ceres. Please follow the instructions in the [docs](https://docs.pupil-labs.com/#windows-dependencies) __very__ closely. Setting up the development environment on Windows is much more involved in comparison to development environment on macOS or Linux.

**Comment by injoonhong** [comment]
Thank you for your kind explanation. Unfortunately, I am using x64 Native Tools Command Prompt for VS 2017 Preview. It is good to know that 3d detector is related to the ceres library. 

I have tried to follow docs instructions very closely. Is this problem somehow related to boost library? When I built the boost library 1.64 for msvc 14.1 x64, I have seen warning msgs which is "unknown compiler version - please run the configure tests and report the results". Even though I got this msgs, I can successfully build the boost python library. Other than this msgs, I do not have any any errors.

**Comment by injoonhong** [comment]
@willpatera I have just rebuild ceres library and tried to build the detector 3d one more time again, but not succeed it.

**Comment by injoonhong** [comment]
@willpatera  After I added #define BOOST_ALL_NO_LIB in a source file, I can resolve the vc141 error msg. 

However,  I still get an issue with detector_3d.obj compiled with older version. I successfully did upgrade the project from VS 2015 to VS 2017 preview and built the library without any problems. I don't know why this problem still happens. 

Can somebody can upload the ceres_static.lib, libglog_static.lib compiled in VS2017 version so that I can give it a try?
